components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -543,16 +543,27 @@ internal sealed class SubscriptionWrapper : ISubscription {
                         if (change.Key == null) {
                             continue;
                         }
-                        _logger.Information("Set Monitoring to {value} for {count} nodes in subscription " +
-                            "{subscription}", change.Key.Value, change.Count(),
-                            rawSubscription.DisplayName);
-                        var results = rawSubscription.SetMonitoringMode(change.Key.Value,
-                            change.Select(t => t.Item).ToList());
+                        _logger.Information("Set monitoring to \"{value}\" for {count} items in subscription \"{subscription}\".",
+                            change.Key.Value, change.Count(), rawSubscription.DisplayName);
+
+                        var itemsToChange = change.Select(t => t.Item).ToList();
+                        var results = rawSubscription.SetMonitoringMode(change.Key.Value, itemsToChange);
                         if (results != null) {
-                            _logger.Information("Failed to set monitoring for {count} nodes in subscription " +
-                                "{subscription}",
-                                results.Count(r => (r == null) ? false : StatusCode.IsNotGood(r.StatusCode)),
-                                rawSubscription.DisplayName);
+                            var erroneousResultsCount = results
+                                .Count(r => (r == null) ? false : StatusCode.IsNotGood(r.StatusCode));
+
+                            // Check the number of erroneous results and log.
+                            if (erroneousResultsCount > 0) {
+                                _logger.Warning("Failed to set monitoring for {count} items in subscription \"{subscription}\".",
+                                    erroneousResultsCount, rawSubscription.DisplayName);
+
+                                for (int i = 0; i < results.Count && i < itemsToChange.Count; ++ i) {
+                                    if (StatusCode.IsNotGood(results[i].StatusCode)){
+                                        _logger.Warning("Set monitoring for item \"{item}\" in subscription \"{subscription}\" failed with \"{status}\".",
```

Comment:
Consider switching to `_logger.Debug()` when merging to `main` as we might have several thousands of items.

Reply:
I think this is an abnormal situation therefore a warning is appropriate here.

Reply:
OK. Then we can leave it as it is.